### PR TITLE
chore(dev): add pre-commit hooks

### PR DESCRIPTION
**Pull request by mertalev**
## Description

This PR adds a number of pre-commit hooks and configures Renovate to automatically update them. Besides being a general DX improvement, the hope is that Renovate will use the `poetry-lock` hook to produce correct lock files for ML.

### CODE REVIEW

**Comment by bo0tzz** [comment]
Is this expected to fix errors such as https://github.com/immich-app/immich/pull/5180#issuecomment-1818435984 ?

File: .pre-commit-config.yaml
Comment (jrasm91) [inline .pre-commit-config.yaml:1]: Everything else uses `.yml` instead of `.yaml` can this file as well, or does renovate expect `yaml`?

**Review by jrasm91 — APPROVED** [review]

File: .pre-commit-config.yaml
Comment (mertalev) [reply .pre-commit-config.yaml:1]: It actually needs to be `.yaml` since this is what the git hook itself looks for.

**Comment by jrasm91** [comment]
This didn't actually resolve the poetry lock problem and I'm so-so on these pre-commit hooks in general. Feel free to reopen if you still want to implement something like this.

**Pull request closed without merging**